### PR TITLE
10 ocr synthetic data

### PR DESCRIPTION
**Pull request by eddableheath**
This PR adds OCR synthetic generation to the dataloader found in `multieurlex_utils.py`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,29 +27,11 @@ classifiers = [
...
-  "librosa",
-  "torch",
-  "torcheval",
-  "pillow",
-  "unbabel-comet",
-  "accelerate",
-  "tensorboard",
-  "scikit-learn"
-]
+dynamic = ["dependencies", "optional-dependencies"]
```
Comment (lannelin) [inline pyproject.toml:30]: i've changed this to use requirements now

File: src/arc_spice/data/multieurlex_utils.py [hunk 2]
```diff
@@ -179,14 +192,36 @@ def load_multieurlex(
...
     dataset_dict, meta_data = load_multieurlex(
         data_dir=data_dir, level=level, languages=langs, drop_empty=drop_empty
     )
     # instantiate the preprocessor
     preprocesser = TranslationPreProcesser(lang_pair)
     # preprocess each split
-    return dataset_dict.map(preprocesser, remove_columns=["text"]), meta_data
+    dataset_dict = dataset_dict.map(preprocesser, remove_columns=["text"])
+
+    # TODO allow for OCR standalone?
```
Comment (lannelin) [inline src/arc_spice/data/multieurlex_utils.py:211]: this is outstanding but perhaps out of scope for this PR? do we want to be able to do this OCR standalone (i.e. not care about source/target stuff that's primarily for translation)?

File: tests/test_multieurlex_utils.py [hunk 3]
```diff
@@ -53,46 +63,223 @@ def test_extract_articles_multi_lang():
...
+            _check_keys_untouched(
+                original_dataset=ds[split],
+                dataset=dataset_dict[split],
+                indices_kept=expected_non_empty_indices,
+                ignore_keys=["text"],
+            )
 
 
-def test_load_multieurlex_for_translation():
+def test_load_multieurlex_multi_lang():
```
Comment (lannelin) [inline tests/test_multieurlex_utils.py:130]: I realised that the previous tests didn't actually cover everything. It now checks that other keys exist. There are alsso separate tests now for the `load_multieurlex_for_pipeline` method. Including one that checks the OCR output

**Comment by eddableheath** [comment]
LGTM, all tests passing on my machine.

**Review by J-Dymond — APPROVED** [review]
All looks good!

**Pull request merged**